src/ConstructiveSolidGeometry/SurfacePrimitives/Rectangle.jl

```diff
@@ -226,4 +226,32 @@ function get_vertices(r::RectangleZ{T}) where {T}
     CartesianPoint{T}(lMax, wMin, r.loc),
     CartesianPoint{T}(lMin, wMax, r.loc),
     CartesianPoint{T}(lMax, wMax, r.loc))
+end
+
+_get_rectangle_coordinates(point::CartesianPoint, r::RectangleX) = (point.y, point.z, point.x)
+_get_rectangle_coordinates(point::CartesianPoint, r::RectangleY) = (point.x, point.z, point.y)
+_get_rectangle_coordinates(point::CartesianPoint, r::RectangleZ) = (point.x, point.y, point.z)
+
+_get_rectangle_coordinates(point::CylindricalPoint, r::Rectangle) = _get_rectangle_coordinates(CartesianPoint(point), r)
+
+function distance_to_surface(point::AbstractCoordinatePoint{T}, r::Rectangle{<:Any, T})::T where {T}
+    l, w, d = _get_rectangle_coordinates(point, r)
+    lMin::T, lMax::T = get_l_limits(r)
+    wMin::T, wMax::T = get_w_limits(r)
+    if l in lMin..lMax && w in wMin..wMax
```

Comment:
Can we use existing internal functions (similar to `_in_planar_u`)?
If `r.l` and `r.w` are `Intervals`, we would determine the edges and create a "new" interval from them.

Reply:
3 options. 

- `_in_planar_u` would work but then you have to create PlanarPoints
- use multiple dispatch. This would be a lot of unnecessary code
- leave as is. Creates new intervals. 

All have drawbacks.